### DISCUSSION TITLE
Vercel does not let me deploy my NextJS app because of the root directory

### DISCUSSION THREAD

**Discussion by simerlec** [vercel/community#1404] [Help] [answered]
Hello,

when I try to deploy a NextJS app it shows me the following error
![image](https://user-images.githubusercontent.com/5793380/215724751-0e62ce09-792f-4064-9720-c4ec4ef2f5bc.png)

but there's no way to set the root directory to something else? If I try to edit the path and check the dropdown it realizes the project is NextJS but only changes the path from `.` to `./` which also shows the same error.

![image](https://user-images.githubusercontent.com/5793380/215725906-f7bce613-92a1-4985-a675-aeb8c73565c0.png)

I'm sure this is a trivial issue but I cannot figure out what I'm supposed to do? 😄 As a workaround I changed to a subfolder, which let me at least deploy a broken build, then went to the project settings afterwards and changed the settings there.

thanks

**Comment by amyegan** [comment]
Thanks for reporting this problem @simerlec. It sounds like you got it working after a few extra steps, but that extra work is definitely not the experience we want anyone to have. I'm sharing this with the team so we can get it fixed for future projects 🙇

**Comment by agadzik** [comment]
hey @simerlec 👋 

I was trying to reproduce this on my end but I've been unsuccessful in my attempts.  Here's what I've tried so far,

- Import a NextJS app, accept default settings and click Deploy - was able to deploy successfully
- Import a NextJS app, change the root directory to something else (pages in this case), then back to the root directory.  Clicked on Deploy and was able to deploy successfully

Are you able to share any additional information with us to help us track down the issue?

**Comment by simerlec** [comment, marked as answer]
hey @amyegan and @agadzik,

thanks for the reply.

I just tried it out with another NextJS project and it seems to work fine. Apparently this error only occurs for this specific project? Not sure what the cause could be. The project is private and has a (not very complex) next config not sure if that influences anything. Everything else looks identical to a project that deploys without issue 🤔 

Sorry that I cannot provide more information, if you have more specific questions, please let me know.

**Reply by agadzik** [reply]
Our framework detection relies on information in the `package.json` file - are you able to share the contents of that file here for the private repo?  If not, only really care about the `dependencies` and `devDependencies` sections if possible 🤞

**Reply by simerlec** [reply]
Here you go 
```
"dependencies": {
    "@babel/runtime": "^7.18.9",
    "@headlessui/react": "^1.4.1",
    "@mapbox/rehype-prism": "^0.7.0",
    "@monaco-editor/react": "4.4.2",
    "@next/font": "13.1.0",
    "@react-spring/web": "^9.3.0",
    "@sentry/nextjs": "^7.14.1",
    "@storybook/builder-webpack5": "^6.5.16",
    "@storybook/manager-webpack5": "^6.5.16",
    "@typeform/embed-react": "^2.1.0",
    "@types/gtag.js": "^0.0.12",
    "@types/nprogress": "^0.2.0",
    "@types/testing-library__jest-dom": "^5.14.5",
    "@vercel/analytics": "^0.1.2",
    "@xstate/react": "^2.0.0",
    "console-feed": "^3.4.3",
    "css-loader": "^6.7.1",
    "dayjs": "^1.10.7",
    "emmet-monaco-es": "5.1.0",
    "firebase": "^8.8.0",
    "firebase-admin": "^9.11.0",
    "fs-extra": "^10.1.0",
    "googleapis": "^91.0.0",
    "he": "^1.2.0",
    "js-beautify": "^1.14.0",
    "jshint": "^2.13.1",
    "lodash": "^4.17.21",
    "mixpanel-browser": "^2.41.0",
    "monaco-editor": "0.33.0",
    "next": "13.1.0",
    "next-firebase-auth": "^0.14.0",
    "node-ical": "^0.14.1",
    "nprogress": "^0.2.0",
    "prismjs": "^1.24.1",
    "react": "^18.2.0",
    "react-beautiful-dnd": "^13.1.0",
    "react-dom": "^18.2.0",
    "react-hot-toast": "^2.1.1",
    "react-query": "3.39.2",
    "react-scroll": "^1.8.4",
    "react-split": "^2.0.12",
    "react-tooltip": "^4.2.21",
    "react-transition-group": "^4.4.2",
    "rehype-react": "^6.2.1",
    "remark-react": "^8.0.0",
    "remark-rehype": "^8.1.0",
    "smoothscroll-polyfill": "^0.4.4",
    "style-loader": "^3.3.1",
    "url": "^0.11.0",
    "webpack": "^5.74.0",
    "xml-js": "^1.6.11",
    "xstate": "^4.30.0"
  },
  "devDependencies": {
    "@babel/core": "^7.15.5",
    "@storybook/addon-actions": "^6.5.16",
    "@storybook/addon-essentials": "^6.5.16",
    "@storybook/addon-links": "^6.5.16",
    "@storybook/addon-postcss": "^2.0.0",
    "@storybook/react": "^6.5.16",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^13.4.0",
    "@types/he": "^1.1.2",
    "@types/htmlhint": "^0.9.3",
    "@types/jest": "^26.0.24",
    "@types/js-beautify": "^1.13.2",
    "@types/lodash.debounce": "^4.0.6",
    "@types/mixpanel-browser": "^2.35.7",
    "@types/node": "^16.7.10",
    "@types/prismjs": "^1.16.6",
    "@types/react": "^18.0.26",
    "@types/react-beautiful-dnd": "^13.1.2",
    "@types/react-scroll": "^1.8.3",
    "@types/react-transition-group": "^4.4.2",
    "@types/smoothscroll-polyfill": "^0.3.1",
    "@typescript-eslint/eslint-plugin": "^4.30.0",
    "@typescript-eslint/parser": "^4.30.0",
    "@xstate/inspect": "^0.6.2",
    "autoprefixer": "^10.4.4",
    "babel-jest": "^27.1.0",
    "babel-loader": "^8.2.2",
    "danger": "^10.6.6",
    "dotenv": "^16.0.1",
    "eslint": "^7.30.0",
    "eslint-config-next": "^11.1.2",
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-jsx-a11y": "^6.4.1",
    "eslint-plugin-react": "^7.25.1",
    "extract-zip": "^2.0.1",
    "htmlhint": "^0.15.1",
    "husky": "^7.0.2",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^27.1.0",
    "jest-watch-typeahead": "^0.6.4",
    "lint-staged": "^11.0.0",
    "msw": "^0.39.2",
    "msw-storybook-addon": "^1.7.0",
    "node-fetch": "^3.2.4",
    "postcss": "^8.4.12",
    "postcss-loader": "^7.0.1",
    "prettier": "^2.7.1",
    "prettier-plugin-tailwindcss": "^0.1.13",
    "remark-parse": "^10.0.1",
    "storybook-addon-next": "^1.7.1",
    "storybook-addon-next-router": "^4.0.2",
    "tailwindcss": "^3.1.8",
    "timezone-mock": "^1.3.4",
    "tsconfig-paths-webpack-plugin": "^3.5.2",
    "typescript": "^4.4.2",
    "unified": "^10.1.2",
    "whatwg-fetch": "^3.6.2"
  },
```

**Reply by agadzik** [reply]
Interesting I see `"next": "13.1.0",` listed so our detection logic should have worked 🤔 

when you're on `https://vercel.com/new`, does it show the NextJS framework logo next to the repo name?  For example,

<img width="339" alt="image" src="https://user-images.githubusercontent.com/1278714/216108901-380cf31c-0a5f-4017-b6df-d86f82ad39b9.png">

If not, can you open up DevTools (on Chrome / Firefox / or Safari) and send me the response of `/api/v1/detect-eligible-projects` from the Network tab?

**Reply by simerlec** [reply]
This is what I see on `https://vercel.com/new/<teamname>`
![image](https://user-images.githubusercontent.com/5793380/216125412-7c72997d-cec5-4bb0-8789-07d4be7b2d48.png)

here's the relevant `/api/v1/detect-eligible-projects` response

```
{
  "projects": [
    {
      "framework": {
        "name": "Next.js",
        "slug": "nextjs",
        "demo": "https://nextjs-template.vercel.app",
        "logo": "https://api-frameworks.vercel.sh/framework-logos/next.svg",
        "darkModeLogo": "https://api-frameworks.vercel.sh/framework-logos/next-dark.svg",
        "screenshot": "https://assets.vercel.com/image/upload/v1673027027/front/import/nextjs.png",
        "tagline": "Next.js makes you productive with React instantly — whether you want to build static or dynamic sites.",
        "description": "A Next.js app and a Serverless Function API.",
        "website": "https://nextjs.org",
        "sort": 1,
        "envPrefix": "NEXT_PUBLIC_",
        "useRuntime": {
          "src": "package.json",
          "use": "@vercel/next"
        },
        "detectors": {
          "every": [
            {
              "matchPackage": "next"
            }
          ]
        },
        "settings": {
          "installCommand": {
            "placeholder": "`yarn install`, `pnpm install`, or `npm install`"
          },
          "buildCommand": {
            "placeholder": "`npm run build` or `next build`",
            "value": "next build"
          },
          "devCommand": {
            "value": "next dev --port $PORT",
            "placeholder": "next"
          },
          "outputDirectory": {
            "placeholder": "Next.js default"
          }
        },
        "recommendedIntegrations": [
          {
            "id": "oac_5lUsiANun1DEzgLg0NZx5Es3",
            "dependencies": [
              "next-plugin-sentry",
              "next-sentry-source-maps"
            ]
          }
        ]
      },
      "name": "<private_repo_project_name>",
      "path": "/.",
      "settings": {
        "buildCommand": null,
        "outputDirectory": null,
        "installCommand": null
      }
    }
  ],
  "workspaceManager": "pnpm"
}
```

**Reply by agadzik** [reply]
Oh! I noticed that the api also detected that you're using a pnpm workspace,

Mind sharing the contents of `pnpm-workspace.yml`

Is this project part of a monorepo or no?

**Reply by simerlec** [reply]
Good point, this is an oversight on my part. We're using pnpm instead of npm but are not making use of workspaces and the project is not part of a monorepo. This file was probably created by default during the switch and has never been actively used 🤔 I'm guessing this could be the root of the problem?

**Reply by agadzik** [reply]
Thats my hunch yeah, also an edge case for us that we can fix in code!  

Try removing that file from your repo, and re-import your project into Vercel 🤞

**Reply by simerlec** [reply]
Just reproduced this issue by switching a previously working NextJS project over to `pnpm` including a `pnpm-lock.yaml` and `pnpm-workspace.yaml`.

Thanks for figuring this out with me @agadzik 🙏
